### PR TITLE
docs: addition of method documentation

### PR DESCRIPTION
**Pull request by JavierDelgadoFernandez**
* Improves method documentation. (addresses #23)

Signed-off-by: Javier Delgado <javier.delgado.fernandez@cern.ch>

### CODE REVIEW

File: invenio_records_files/utils.py [hunk 1]
```diff
@@ -40,7 +45,13 @@ def sorted_files_from_bucket(bucket, keys=None):
 
 
 def record_file_factory(pid, record, filename):
-    """Get file from a record."""
+    """Get file from a record.
+
+    :param pid: TODO: Not used.
```
Comment (JavierDelgadoFernandez) [inline invenio_records_files/utils.py]: This parameter (``pid``) is not used.

Comment (jirikuncar) [inline invenio_records_files/utils.py]: We can call it `dummy` or `_` and in docstring say that it's just to keep function signature.

File: invenio_records_files/__init__.py [hunk 2]
```diff
@@ -135,7 +135,7 @@
 >>> print(record.files['hello.txt']['filetype'])
 txt
 
-Certain key names are however reserved cannot be used for **setting** metadata:
+Certain key names are horaisewever reserved cannot be used for **setting** metadata:
```
Comment (jirikuncar) [inline invenio_records_files/__init__.py]: `horaisewever` ?

File: invenio_records_files/api.py [hunk 3]
```diff
@@ -85,10 +114,22 @@ def dumps(self):
 
 
 def _writable(method):
-    """Check that record is in defined status."""
+    """Check that record is in defined status.
+
+    :param method: Method to be decorated.
+    :returns:
```
Comment (jirikuncar) [inline invenio_records_files/api.py]: Decorator ...

File: invenio_records_files/api.py [hunk 4]
```diff
@@ -112,34 +158,59 @@ def __init__(self, record, bucket=None, file_cls=None):
 
     @property
     def keys(self):
-        """Return file keys."""
+        """Return file keys.
+
+        :returns: Key list of the files files.
```
Comment (jirikuncar) [inline invenio_records_files/api.py]: @JavierDelgadoFernandez can you rephrase this line? `List of file identifiers/names`?

File: invenio_records_files/api.py [hunk 5]
```diff
@@ -112,34 +158,59 @@ def __init__(self, record, bucket=None, file_cls=None):
...
+        """
         return ObjectVersion.get_by_bucket(
             self.bucket).filter_by(key=str(key)).count()
 
     def __getitem__(self, key):
-        """Get a specific file."""
+        """Get a specific file.
+
+        :param key: File key which holds the file.
+        :returns: Instance of file_cls containing the file.
```
Comment (jirikuncar) [inline invenio_records_files/api.py]: Can you try to reference the `file_cls` property?
```
:py:data:`self.file_cls`
```

File: invenio_records_files/api.py [hunk 6]
```diff
@@ -85,10 +114,22 @@ def dumps(self):
...
-        """Send record for indexing."""
+        """Send record for indexing.
+
+        :param args: List of arguments.
+        :param kwargs: Dictionary of arguments for the function.
+        :returns: Execution result of the decorated method.
+
+        :raises InvalidOperationError: It occurs when the bucket is locked or
+            deleted.
+        """
```
Comment (JavierDelgadoFernandez) [inline invenio_records_files/api.py]: I know that this documentation is not showed (Actually not even the ``_writable`` method), but if someone is reviewing the code, it will be helpful.

File: invenio_records_files/api.py [hunk 7]
```diff
@@ -41,39 +41,68 @@ class FileObject(object):
...
         self.obj = obj
         self.data = data
 
     def get_version(self, version_id=None):
-        """Return specific version ``ObjectVersion`` instance or HEAD."""
+        """Return specific version ``ObjectVersion`` instance or HEAD.
+
+        :param version_id: Version ID of the object.
+        :returns: ObjectVersion instance or HEAD of the stored object.
```
Comment (jirikuncar) [inline invenio_records_files/api.py]:
```
:class:`~... ObjectVersion`
```

File: invenio_records_files/utils.py [hunk 8]
```diff
@@ -40,7 +45,13 @@ def sorted_files_from_bucket(bucket, keys=None):
 
 
 def record_file_factory(pid, record, filename):
-    """Get file from a record."""
+    """Get file from a record.
+
+    :param pid: Not used. It keeps the function signature.
+    :param record: Record which contains the files.
+    :param filename: Name of the file to be returned.
+    :returns: File object or None if not found.
```
Comment (jirikuncar) [inline invenio_records_files/utils.py]:
```
``None``
```

File: invenio_records_files/utils.py [hunk 9]
```diff
@@ -31,7 +31,12 @@
 
 
 def sorted_files_from_bucket(bucket, keys=None):
-    """Return files from bucket sorted by given keys."""
+    """Return files from bucket sorted by given keys.
+
+    :param bucket: ``Bucket`` containing the files.
```
Comment (jirikuncar) [inline invenio_records_files/utils.py]:
```
:class:`~invenio_records_files.models:Bucket`
```

**Comment by jirikuncar** [comment]
ping @JavierDelgadoFernandez

**Comment by JavierDelgadoFernandez** [comment]
@jirikuncar Updated.

File: docs/conf.py [hunk 10]
```diff
@@ -326,6 +326,11 @@
...
 #texinfo_no_detailmenu = False
 
-
 # Example configuration for intersphinx: refer to the Python standard library.
-intersphinx_mapping = {'https://docs.python.org/': None}
+
+intersphinx_mapping = {
+    'https://docs.python.org/': None,
+    'invenio-files-rest': (
+        'http://pythonhosted.org/invenio-files-rest', None
```
Comment (jirikuncar) [inline docs/conf.py]: https

File: invenio_records_files/api.py [hunk 11]
```diff
@@ -41,39 +41,69 @@ class FileObject(object):
...
+        """
         self.obj = obj
         self.data = data
 
     def get_version(self, version_id=None):
-        """Return specific version ``ObjectVersion`` instance or HEAD."""
+        """Return specific version ``ObjectVersion`` instance or HEAD.
+
+        :param version_id: Version ID of the object.
+        :returns: :class:`~invenio_files_rest.models.ObjectVersion` instance or
```
Comment (jirikuncar) [inline invenio_records_files/api.py]: 👍

File: invenio_records_files/api.py [hunk 12]
```diff
@@ -85,10 +115,22 @@ def dumps(self):
...
+
+    :param method: Method to be decorated.
+    :returns: Function decorated.
+    """
     @wraps(method)
     def wrapper(self, *args, **kwargs):
-        """Send record for indexing."""
+        """Send record for indexing.
+
+        :param args: List of arguments.
```
Comment (jirikuncar) [inline invenio_records_files/api.py]: IMHO it is not necessary to document internal `*args`, `**kwargs`. It doesn't add any value.

File: invenio_records_files/api.py [hunk 13]
```diff
@@ -224,18 +316,24 @@ class FilesMixin(object):
     """
 
     file_cls = FileObject
+
```
Comment (jirikuncar) [inline invenio_records_files/api.py]: what about adding a docstring here instead of just an empty line?

File: invenio_records_files/api.py [hunk 14]
```diff
@@ -224,18 +316,24 @@ class FilesMixin(object):
...
     files_iter_cls = FilesIterator
 
     def _create_bucket(self):
         """Return an instance of ``Bucket`` class.
 
         .. note:: Reimplement in children class for custom behavior.
+
+        :returns: Instance of ``Bucket``.
```
Comment (jirikuncar) [inline invenio_records_files/api.py]: `:class:...`

File: invenio_records_files/api.py (hunk 13 above)
Comment (JavierDelgadoFernandez) [inline invenio_records_files/api.py]: I was not adding it because by default it is generating an "alias of class" comment with the correct link. Should I add a custom comment?
aliasof

**Comment by JavierDelgadoFernandez** [comment]
@jirikuncar Updated.

File: invenio_records_files/models.py [hunk 15]
```diff
@@ -44,20 +44,31 @@ class RecordsBuckets(db.Model):
...
+    """It is used by SQLAlchemy for optimistic concurrency control."""
 
     @classmethod
     def create(cls, record, bucket):
-        """Create a new RecordsBuckets and adds it to the session."""
+        """Create a new RecordsBuckets and adds it to the session.
+
+        :param record: Record used to relate with the ``Bucket``.
+        :param bucket: Bucket used to relate with the ``Record``.
+        :returns: The :class:`invenio_records_files.models.RecordsBuckets`
```
Comment (jirikuncar) [inline invenio_records_files/models.py]: You can use `~` to show only class name in generated docs.

**Comment by JavierDelgadoFernandez** [comment]
@jirikuncar Updated.
